lib/portlayer/storage/vsphere/volume.go

```diff
@@ -189,8 +203,26 @@ func (v *VolumeStore) VolumeCreate(ctx context.Context, ID string, store *url.UR
 	return vol, nil
 }
 
-func (v *VolumeStore) VolumeDestroy(ctx context.Context, ID string) error {
-	return fmt.Errorf("TBD.  Not supported yet")
+func (v *VolumeStore) VolumeDestroy(ctx context.Context, vol *storage.Volume) error {
+	if err := volumesInUse(ctx, vol.ID); err != nil {
```

Comment:
@fdawg4l 
vsphere will only prevent the deletion of the disk if the containerVM is powered on (iirc). You will get an error during creation if the volume disk doesn't exist, which is the desired behaviour.

The case we may not address here is:
1. call volumesInUse - no containers
2. containerVM created referencing volume
3. volume deleted
4. containerVM powered on - will fail due to missing disk

One approach is to attempt _pressimistic_ reference counting in the PL:
1. set a reference count on initialization having checked existing containers
2. each time volumejoin is called, increment
3. check for zero reference count on rm
4. allow force deletion to bypass inconsistency.

There are a couple of other approaches that could reduce, but not close, this hole. My inclination at this time is to provide a good error message for the failing power on and then leverage the `deletable` flag once we have disklib access (or FCD appears).